driver/src/test/java/org/neo4j/driver/internal/connector/socket/SocketClientTest.java

```diff
@@ -26,14 +27,15 @@
 import java.net.ServerSocket;
 
 import org.neo4j.driver.exceptions.ClientException;
-import org.neo4j.driver.internal.logging.DevNullLogger;
 
 public class SocketClientTest
 {
     @Rule
     public ExpectedException exception = ExpectedException.none();
 
-    @Test
+    // TODO: This is not possible with blocking NIO channels, unless we use inputStreams, but then we can't use
+    // off-heap buffers. We need to swap to use selectors, which would allow us to time out.
+    @Test @Ignore
```

Comment:
Well, there is a trick http://stackoverflow.com/questions/2866557/timeout-for-socketchannel-doesnt-work, but this is not so important for now.